File: notion/commands/comment.py
"""Comment commands."""
from __future__ import annotations
import json

import click

from notion.client import NotionClient, _normalise_id
from notion.errors import NotionCliError, handle_error
from notion.output import print_json, print_ids


@click.group()
def comment():
    """Comment commands."""
    pass


@comment.command("add")
@click.argument("page_id")
@click.argument("text")
def add(page_id, text):
    """Add a comment to a page."""
    try:
        pid = _normalise_id(page_id)
        client = NotionClient()
        result = client.post("/comments", {
            "parent": {"page_id": pid},
            "rich_text": [{"type": "text", "text": {"content": text}}]
        })
        client.close()
        print_json(result)
    except NotionCliError as e:
        handle_error(e)


@comment.command("list")
@click.argument("page_id")
@click.option("--output", type=click.Choice(["json", "ids"]), default="json")
def list_comments(page_id, output):
    """List comments on a page."""
    try:
        pid = _normalise_id(page_id)
        client = NotionClient()
        results = list(client.paginate("GET", "/comments", params={"block_id": pid}))
        client.close()

        if output == "ids":
            print_ids(results)
        else:
            print_json(results)
    except NotionCliError as e:
        handle_error(e)

File: src/commands/comment.ts
import { Command } from 'commander'
import { createNotionClient, normaliseId, collectPaginatedAPI } from '../client.js'
import { printJSON, printIds } from '../output.js'

export function registerComment(program: Command): void {
  const comment = program.command('comment').description('Comment commands')

  // comment add
  comment.command('add <page-id> <text>')
    .description('Add a comment to a page')
    .action(async (pageId: string, text: string) => {
      const client = createNotionClient()
      const result = await client.comments.create({
        parent: { page_id: normaliseId(pageId) },
        rich_text: [{ type: 'text', text: { content: text } }],
      }) as any
      printJSON(result)
    })

  // comment list
  comment.command('list <page-id>')
    .description('List comments on a page')
    .option('--output <format>', 'Output format: json|ids', 'json')
    .action(async (pageId: string, opts: { output: string }) => {
      const client = createNotionClient()
      const results = await collectPaginatedAPI(client.comments.list, {
        block_id: normaliseId(pageId),
      })

      if (opts.output === 'ids') {
        printIds(results as Array<{ id: string }>)
      } else {
        printJSON(results)
      }
    })
}
